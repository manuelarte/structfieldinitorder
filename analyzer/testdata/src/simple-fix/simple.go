package main

import (
	"time"
)

type Person struct {
	Name      string
	Surname   string
	Birthdate time.Time
}

func mainSimple() {
	_ = Person{ // want `fields for struct "Person" are not instantiated in order`
		Birthdate: time.Now(),
		Surname:   "Doe",
		Name:      "John",
	}

	_ = &Person{ // want `fields for struct "Person" are not instantiated in order`
		Birthdate: time.Now(),
		Surname:   "Doe",
		Name:      "John",
	}

	_ = &Person{
		Name:      "John",
		Surname:   "Doe",
		Birthdate: time.Now(),
	}
}
