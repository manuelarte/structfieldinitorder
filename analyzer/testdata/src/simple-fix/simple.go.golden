package main

import (
	"time"
)

type Person struct {
	Name      string
	Surname   string
	Birthdate time.Time
}

func mainSimple() {
	_ = Person{ // want `fields for struct "Person" are not instantiated in order`
		Name:      "John",
		Surname:   "Doe",
	    Birthdate: time.Now(),
	}

	_ = &Person{ // want `fields for struct "Person" are not instantiated in order`
		Name:      "John",
		Surname:   "Doe",
		Birthdate: time.Now(),
	}

	_ = &Person{
		Name:      "John",
		Surname:   "Doe",
		Birthdate: time.Now(),
	}
}
